arirs/src/client.rs

```
@@ -2,208 +2,107 @@ use std::time::Duration;
+                    message = ws_receiver.next() => {
+                        match message {
+                            Some(message) => {
+                                let message = message?;
+                                match message {
+                                    tungstenite::Message::Text(_) => {
+                                        if let Err(e) = tx.send(serde_json::from_slice(message.into_data().as_slice()).map_err(|err| AriError::Unknown(err.to_string()))?) {
+                                            event!(Level::ERROR, "Error sending event: {}", e);
+                                        }
+                                    }
+                                    tungstenite::Message::Ping(data) => {
+                                        event!(Level::TRACE, "Received WebSocket Ping, sending Pong");
+                                        ws_sender.send(tungstenite::Message::Pong(data)).await?;
+                                    }
+                                    tungstenite::Message::Pong(_) => {
+                                        event!(Level::TRACE, "Received WebSocket Pong");
+                                    },
+                                    tungstenite::Message::Close(frame) => {
+                                        event!(Level::INFO, "WebSocket closed: {:?}", frame);
+                                        break;
+                                    },
+                                    _ => {
+                                        event!(Level::INFO, "Unknown WebSocket message");
+                                    }
                                 }
                             }
-                        }
-                        None => {
-                            event!(Level::INFO, "WebSocket closed");
-                            break;
+                            None => {
+                                tracing::error!("WebSocket closed");
```

question: is a websocket closing an error?

---

Probably, since we're the ones driving it. I would much rather return an error here. I'll be going over logging soon.